src/MCPClient/lib/clientScripts/sanitize_names.py

```diff
@@ -21,100 +21,74 @@
 # @subpackage MCPClient
 # @author Joseph Perry <joseph@artefactual.com>
 
-import string
 import os
-from shutil import move as rename
-import sys
-import unicodedata
+import re
+import shutil
+
+from scandir import scandir
 from unidecode import unidecode
-from archivematicaFunctions import unicodeToStr
 
-VERSION = "1.10." + "$Id$".split(" ")[1]
-valid = "-_.()" + string.ascii_letters + string.digits
-replacementChar = "_"
+from archivematicaFunctions import strToUnicode
 
+VERSION = "1.10." + "$Id$".split(" ")[1]
 
-def transliterate(basename):
-    # We get a more meaningful name sanitization if UTF-8 names
-    # are correctly decoded to unistrings instead of str
-    try:
-        return unidecode(basename.decode("utf-8"))
-    except UnicodeDecodeError:
-        return unidecode(basename)
+# Letters, digits and a few punctuation characters
+ALLOWED_CHARS = re.compile(r"[^a-zA-Z0-9\-_.\(\)]")
+REPLACEMENT_CHAR = "_"
 
 
-def sanitizeName(basename):
-    ret = ""
-    basename = transliterate(basename)
-    for c in basename:
-        if c in valid:
-            ret += c
-        else:
-            ret += replacementChar
-    return ret.encode("utf-8")
+def sanitize_name(basename):
+    unicode_name = unidecode(strToUnicode(basename))
+    # Handle the case where '' is returned by unidecode
```

Comment:
Does this comment add much? It would be more useful to explain why we don’t want to return an empty string here.

Are there circumstances where `strToUnicode` can return an empty string? What’s the best thing to do in that case?

Reply:
Yeah, on review, that's a pretty useless comment. Good catch.

The return value of this function becomes the new filename, so we don't want to return "" (there are also some assumptions made when checking for an existing file that will break down).

[strToUnicode](https://github.com/artefactual/archivematica/blob/qa/1.x/src/archivematicaCommon/lib/archivematicaFunctions.py#L93) shouldn't return an empty string unless it's passed one. If that is happening, things are pretty broken, but I've added a ValueError for that case.

The if statement this comment is referring to handles the case where there is nothing in the filename that unidecode recognizes, e.g. if the filename is just an emoji: `🚀`. That's obviously a pretty rare case — the old function didn't bother handling it.

I've updated with a more detailed comment, let me know if it's clear.